### ISSUE TITLE
Thoughts about Option.get

### ISSUE THREAD

**Issue by mgyucht** [databricks/scala-style-guide#25]
I think we should not use `Option.get` in our code. When the object in question is `None`, the error message is a cryptic `NoSuchElementException: None.get`. Instead, we should use `Option.getOrElse(throw new Exception(exceptionMsg))` so that it is easier to discover where and why the object was `None` in the first place.

**Comment by rxin** [comment]
I tend to agree - the only exception might be when the scope is super local and known for sure there is an element (e.g. with an if check beforehand on option.isDefined)

**Comment by mgyucht** [comment]
Yeah, that makes sense. At least it reduces a level of nesting. Also if the variable is volatile you might run into race conditions between the check and the dereference.

**Comment by hupq22** [comment]
I completely agree with your perspective. Using Option.get can lead to the confusing NoSuchElementException when the value is None, which doesn't provide helpful context for debugging.

In cases where the scope is very localized and you are certain an element exists, it's still better to avoid Option.get. Instead, using Option.getOrElse(throw new Exception(exceptionMsg)) offers clearer error reporting. This way, if the object is None, the thrown exception will include a meaningful message that indicates exactly where and why the failure occurred.

This approach not only improves code readability but also enhances maintainability by making potential issues easier to trace. It's a good practice to prioritize clarity in error handling to streamline debugging processes.

What do you all think? Would you be open to adopting this pattern in our codebase?
